tests/units/anta_tests/test_snmp.py

```diff
@@ -152,4 +152,66 @@
             "messages": ["SNMP contact is not configured."],
         },
     },
+    {
+        "name": "success",
+        "test": VerifySnmpPDUs,
+        "eos_data": [
+            {
+                "counters": {
+                    "inGetPdus": 3,
+                    "inGetNextPdus": 2,
+                    "inSetPdus": 3,
+                    "outGetResponsePdus": 3,
+                    "outTrapPdus": 9,
+                },
+            }
+        ],
+        "inputs": {},
+        "expected": {"result": "success"},
+    },
+    {
+        "name": "success-specific-pdus",
+        "test": VerifySnmpPDUs,
+        "eos_data": [
+            {
+                "counters": {
+                    "inGetPdus": 3,
+                    "inGetNextPdus": 0,
+                    "inSetPdus": 0,
+                    "outGetResponsePdus": 0,
+                    "outTrapPdus": 9,
+                },
+            }
+        ],
+        "inputs": {"pdus": ["inGetPdus", "outTrapPdus"]},
+        "expected": {"result": "success"},
+    },
+    {
+        "name": "failure-counters-not-found",
+        "test": VerifySnmpPDUs,
+        "eos_data": [
+            {
+                "counters": {},
+            }
+        ],
+        "inputs": {},
+        "expected": {"result": "failure", "messages": ["SNMP counter details are not found."]},
+    },
+    {
+        "name": "failure-incorrect-counters",
+        "test": VerifySnmpPDUs,
+        "eos_data": [
+            {
+                "counters": {
+                    "inGetPdus": 0,
+                    "inGetNextPdus": 2,
+                    "inSetPdus": 0,
+                    "outGetResponsePdus": 3,
+                    "outTrapPdus": 9,
+                },
+            }
+        ],
+        "inputs": {},
+        "expected": {"result": "failure", "messages": ["The following SNMP PDU(s) are not found or have zero PDU counter:\n{'inGetPdus': 0, 'inSetPdus': 0}"]},
```

Comment:
please add the case where pdu is not found

Reply:
Added check. Thank you!!